CHANGELOG/CHANGELOG-4.0.md

```diff
@@ -0,0 +1,261 @@
+# Release notes for v4.0.0
+
+[Documentation](https://kubernetes-csi.github.io)
+
+# Changelog since v3.0.0
+
+## Breaking Changes
+
+- Update to client/v4 ([#451](https://github.com/kubernetes-csi/external-snapshotter/pull/451), [@xing-yang](https://github.com/xing-yang))
+
+- Move Snapshot APIs to V1. ([#419](https://github.com/kubernetes-csi/external-snapshotter/pull/419), [@xing-yang](https://github.com/xing-yang))
+
+- Update snapshot-controller, CSI snapshotter sidecar, and validation webhook based on v1 apis. ([#425](https://github.com/kubernetes-csi/external-snapshotter/pull/425), [@xing-yang](https://github.com/xing-yang))
+
+### Feature
+
+- Add `process_start_time_seconds` metric. ([#387](https://github.com/kubernetes-csi/external-snapshotter/pull/387), [@Jiawei0227](https://github.com/Jiawei0227))
+- The CSI snapshotter has a new `--extra-create-metadata` flag.
+  When set to true, the name and namespace of the source VolumeSnapshot, and the name of the source  VolumeSnapshotContent will be passed as parameters to the CSI driver in CreateSnapshotRequest. ([#375](https://github.com/kubernetes-csi/external-snapshotter/pull/375), [@chrishenzie](https://github.com/chrishenzie))
+- The `--http-endpoint` and `--metrics-path` flags have been added to support metrics for the snapshot-controller. To enable metrics, the user must configure a `--metrics-path` such as `:9102`. The default value for `--metrics-path` is the empty string, meaning this feature is disabled by default. The default value for `--metrics-path` is `/metrics`. ([#409](https://github.com/kubernetes-csi/external-snapshotter/pull/409), [@ggriffiths](https://github.com/ggriffiths))
```

Comment:
We may need to merge this with the note for #439 below that changed the flag names and added the deprecation notice.

Reply:
409 is for snapshot-controller while 439 is for the sidecar.  Let me separate them into different categories.

Reply:
Hmmm I think we want snapshot-controller and the sidecar to also use consistent flag names. Can we quickly put together to make them use similar flag names?

Reply:
They do use similar flag names.

Snapshot-controller just introduced metrics support.  So for snapshot-controller, there's no deprecation as it never used metrics-address.  It just directly uses http-endpoint and metrics-path.

For the sidecar,  metrics-address is deprecated and replaced by http-endpoint.  metrics-path is already there and we are not changing it.

Reply:
I modified the changelog.  Hope it is more clear.  PTAL.

Reply:
ah sorry i'm confusing myself with all the flags :-) Thanks for clarifying